tests/unittests/config/test_cc_ssh.py

```diff
@@ -22,80 +25,92 @@
-                f.write(" ".join(key_data))
-
-        cc_ssh.KEY_FILE_TPL = os.path.join(hostkey_tmpdir, "ssh_host_%s_key")
-
-    def test_apply_credentials_with_user(self, m_setup_keys):
-        """Apply keys for the given user and root."""
-        keys = ["key1"]
-        user = "clouduser"
-        cc_ssh.apply_credentials(keys, user, False, ssh_util.DISABLE_USER_OPTS)
-        self.assertEqual(
-            [
-                mock.call(set(keys), user),
-                mock.call(set(keys), "root", options=""),
-            ],
-            m_setup_keys.call_args_list,
-        )
-
-    def test_apply_credentials_with_no_user(self, m_setup_keys):
-        """Apply keys for root only."""
-        keys = ["key1"]
-        user = None
-        cc_ssh.apply_credentials(keys, user, False, ssh_util.DISABLE_USER_OPTS)
-        self.assertEqual(
-            [mock.call(set(keys), "root", options="")],
-            m_setup_keys.call_args_list,
-        )
 
-    def test_apply_credentials_with_user_disable_root(self, m_setup_keys):
-        """Apply keys for the given user and disable root ssh."""
-        keys = ["key1"]
-        user = "clouduser"
-        options = ssh_util.DISABLE_USER_OPTS
-        cc_ssh.apply_credentials(keys, user, True, options)
-        options = options.replace("$USER", user)
-        options = options.replace("$DISABLE_USER", "root")
-        self.assertEqual(
-            [
-                mock.call(set(keys), user),
-                mock.call(set(keys), "root", options=options),
-            ],
-            m_setup_keys.call_args_list,
-        )
+@mock.patch(MODPATH + "ssh_util.setup_user_keys")
+class TestHandleSsh:
+    """Test cc_ssh handling of ssh config."""
 
-    def test_apply_credentials_with_no_user_disable_root(self, m_setup_keys):
-        """Apply keys no user and disable root ssh."""
-        keys = ["key1"]
-        user = None
+    @pytest.mark.parametrize(
+        "args,kwargs",
+        [
+            # For the given user and root.
+            pytest.param([["key1"], "clouduser", False], {}, id="with_user"),
+            # For root only.
+            pytest.param([["key1"], None, False], {}, id="with_no_user"),
+            # For the given user and disable root ssh.
+            pytest.param(
+                [
+                    ["key1"],
+                    "clouduser",
+                    True,
+                ],
+                {},
+                id="with_user_disable_root",
+            ),
+            # No user and disable root ssh.
+            pytest.param(
+                [
+                    ["key1"],
+                    None,
+                    True,
+                ],
+                {},
+                id="with_no_user_disable_root",
+            ),
+        ],
+    )
+    def test_apply_credentials(
+        self,
+        m_setup_keys,
+        args,
+        kwargs,
+    ):
+        keys, user, *args, disable_root_opts = args
```

Comment:
Why the `*args` here? I think this call should just be `keys, user, disable_root` = args

Reply:
Same as in [r872517535](https://github.com/canonical/cloud-init/pull/1451/files/4252a5f6f0ca225afb78da86da7cbaeb4f1c6018#r872517535). Simplified.